openfisca_web_api/handlers.py

```diff
@@ -28,10 +25,22 @@ def calculate(tax_benefit_system, input_data):
             entity_result = str(result[entity_index])
         else:
             entity_result = result.tolist()[entity_index]
-
+        # Don't use dpath if we have a key that is numeric,
+        # mandatory for dpath upgrade, see https://github.com/dpath-maintainers/dpath-python/issues/160
+        if period.isnumeric():
+            if computation_results == {}:
+                computation_results = {entity_plural: {entity_id: {variable_name: {period: entity_result}}}}
+            else:
+                if entity_plural in computation_results and entity_id in computation_results[entity_plural]:
+                    if variable_name in computation_results[entity_plural][entity_id]:
+                        computation_results[entity_plural][entity_id][variable_name][period] = entity_result
+                    else:
+                        computation_results[entity_plural][entity_id] = {variable_name: {period: entity_result}}
+                else:
+                    computation_results[entity_plural] = {entity_id: {variable_name: {period: entity_result}}}
+            continue
```

Comment:
@benoit-cty When I comment this code, the `test_two_periods` ends without error while its `simulation_json` contains numeric periods (strings containing yearly periods). 🤔 
What error are we solving here? 🙂

Reply:
Sorry, you have to do a `pip install dpath=2.5.0` to have the problem.

Reply:
Sorry, I mixed my virtual environments 😬 
So, with dpath update to 2.0.5, we get None values for variables at yearly periods as they are seen as numeric periods (even if they are given as string to dpath.util.new function). 

For example before this PR fix, `pytest tests/web_api/test_calculate.py` had this result:
```shell
FAILED tests/web_api/test_calculate.py::test_periods - AssertionError: assert [None, None, ...ne, None, ...] == {'2017': 200.0}
FAILED tests/web_api/test_calculate.py::test_two_periods - AssertionError: assert [None, None, ...ne, None, ...] == {'2017': 200.0, '2018': 200.0}
```